pulseapi/entries/views.py

```diff
@@ -61,6 +62,31 @@ def toggle_bookmark(request, entryid):
 
 
 @api_view(['PUT'])
+def toggle_featured(request, entryid):
```

Comment:
It is now worth doing a little refactoring: we have two functions that are effectively identical except for the model value they update: this is the perfect time to create a single function that does the bulk of the work, and two functions that rely on that but house the "different" bits of code, so that we get something like:

```python
def get_entry_to_modify(request, entryid):
    '''
    This function will return an Entry object if the user
    associated with the request has entry change rights,
    or it will return a (non-200) Response object to be
    sent back as server response when they do not.
    '''
    
    # all the code that checks user's permissions
    # and either fetches an entry or forms a non-2xx
    # Response goes here

@api_view(['PUT'])
def toggle_featured(request, entryid):
    entry = get_entry_to_modify(request, entryid)

    if isInstance(entry, Response) is True:
        return target

    entry.featured = not entry.featured
    entry.save()

    return Response("Toggled featured status.", status=status.HTTP_204_NO_CONTENT)

@api_view(['PUT'])
 def toggle_moderation(request, entryid, stateid):
    entry = get_entry_to_modify(request, entryid)

    if isInstance(entry, Response) is True:
        return target

    entry.moderation_state = ModerationState.objects.get(id=stateid)
    entry.save()

    return Response("Toggled moderation state.", status=status.HTTP_204_NO_CONTENT)
```